src/bridge/trx/index.ts

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { TronWeb } from "tronweb-typings";
+import { AllbridgeCoreClient } from "../../client/core-api";
+import { sleep } from "../../utils/calculation";
+import {
+  ApprovalBridge,
+  ApproveData,
+  GetTokenBalanceData,
+  TransactionResponse,
+  TxSendParams,
+} from "../models";
+import { getNonce } from "../utils";
+
+export const MAX_AMOUNT =
+  "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+
+export class TronBridge extends ApprovalBridge {
+  constructor(public api: AllbridgeCoreClient, public tronWeb: TronWeb) {
+    super();
+  }
+
+  async getTokenBalance(data: GetTokenBalanceData): Promise<string> {
+    const contract = await this.getContract(data.tokenAddress);
+    const balance = await contract.balanceOf(data.account).call();
+    return balance.toString();
+  }
+
+  async sendTx(params: TxSendParams): Promise<TransactionResponse> {
+    const {
+      amount,
+      contractAddress,
+      fromTokenAddress,
+      toChainId,
+      toAccountAddress,
+      toTokenAddress,
+      messenger,
+      fee,
+    } = params;
+
+    const bridgeContract = await this.getContract(contractAddress);
+    const nonce = getNonce().toJSON().data;
+
+    const swapAndBridgeMethod = bridgeContract.methods.swapAndBridge(
+      fromTokenAddress,
+      amount,
+      toAccountAddress,
+      toChainId,
+      toTokenAddress,
+      nonce,
+      messenger
+    );
+
```

Comment:
Do we need to estimate gas/energy here before sending transaction like we do it in the `EvmBridge` ?